src/library/scala/collection/Iterable.scala

```diff
@@ -52,3 +57,19 @@ object Iterable extends TraversableFactory[Iterable] {
 
 /** Explicit instantiation of the `Iterable` trait to reduce class file size in subclasses. */
 abstract class AbstractIterable[+A] extends AbstractTraversable[A] with Iterable[A]
+
+/** PairedIterable for functions on paired iterables */
+class PairedIterable[K, V](x: Iterable[(K, V)]) {
+  def reduceByKey(func: (V,V) => V) = {
```

Comment:
Never heard of `reduceToMap`. And couldn't find it. But it seems to have a map, and then a reduce. I am just having the reduce here. And, you are right. I should add it there, but I had technical difficulties. Anyway, I can do that.
`l.map(x => (x,1)).reduceByKey(_+_)`
Take a look at this:
http://ideone.com/dyrkYM